### PR TITLE
Support for comments on views for Delta Lake

### PR DESCRIPTION
**Pull request by marcinsbd**
## Description
Support for comments on views for Delta Lake

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Add support for comments on views.
```

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/TrinoViewHiveMetastore.java [hunk 1]
```diff
@@ -173,4 +173,19 @@ public Optional<ConnectorViewDefinition> getView(SchemaTableName viewName)
                         view.getParameters(),
                         view.getOwner()));
     }
+
+    public void setViewComment(ConnectorSession session, SchemaTableName viewName, Optional<String> comment)
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/TrinoViewHiveMetastore.java:177]: This method is used only from Delta Lake connector, but it exists in Hive connector. Is it possible to extract `io.trino.plugin.iceberg.catalog.hms.TrinoHiveCatalog#updateViewComment` into here and reuse between Delta Lake and Iceberg?

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java [hunk 2]
```diff
@@ -556,4 +558,42 @@ public void testMaterializedView()
 
         assertUpdate("DROP MATERIALIZED VIEW " + viewName);
     }
+
+    @Test
+    public void testCommentView()
+    {
+        if (hasBehavior(SUPPORTS_CREATE_VIEW) && !hasBehavior(SUPPORTS_COMMENT_ON_VIEW)) {
+            String viewName = "nation_v";
```
Comment (ebyhr) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java:566]: Please note that we avoid using static identifiers because it will lead to flaky tests when the test runs on SaaS connectors.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/TrinoViewHiveMetastore.java [hunk 3]
```diff
@@ -173,4 +173,19 @@ public Optional<ConnectorViewDefinition> getView(SchemaTableName viewName)
...
+        ConnectorViewDefinition newDefinition = new ConnectorViewDefinition(
+                definition.getOriginalSql(),
+                definition.getCatalog(),
+                definition.getSchema(),
+                definition.getColumns(),
+                comment,
+                definition.getOwner(),
+                definition.isRunAsInvoker());
+
+        createView(session, viewName, newDefinition, true);
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/TrinoViewHiveMetastore.java:189]: Calling `createView` method looks weird to me.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java [hunk 4]
```diff
@@ -189,13 +189,17 @@ protected QueryRunner createQueryRunner()
...
+    @SuppressWarnings("DuplicateBranchesInSwitch")
     @Override
     protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
     {
         switch (connectorBehavior) {
             case SUPPORTS_CREATE_VIEW:
                 return true;
 
+            case SUPPORTS_COMMENT_ON_VIEW:
+                return true;
```
[the remark this answers is not in the document]
Comment (marcinsbd) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java:201]: Thanks, good to know.

**Comment by ebyhr** [comment]
/test-with-secrets sha=afa4226e9efea8e38d331e122a95344185fe0432

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (ebyhr) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java:201]: #16825 is already merged. Please rebase and remove this condition.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java [hunk 5]
```diff
@@ -378,21 +378,7 @@ public Table loadTable(ConnectorSession session, SchemaTableName schemaTableName
...
-                definition.getOriginalSql(),
-                definition.getCatalog(),
-                definition.getSchema(),
-                definition.getColumns(),
-                comment,
-                definition.getOwner(),
-                definition.isRunAsInvoker());
-
-        replaceView(session, viewName, view, newDefinition);
+        trinoViewHiveMetastore.updateViewComment(session, viewName, comment);
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java:381]: Please add a preparatory commit to extract a method.

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java [hunk 6]
```diff
@@ -556,4 +558,44 @@ public void testMaterializedView()
...
+
+            // comment deleted
+            assertUpdate("COMMENT ON VIEW " + view.getName() + " IS 'a comment'");
+            assertThat(getTableComment(view.getName())).isEqualTo("a comment");
+            assertUpdate("COMMENT ON VIEW " + view.getName() + " IS NULL");
+            assertThat(getTableComment(view.getName())).isNull();
+        }
+    }
+
+    protected String getTableComment(String tableName)
```
Comment (ebyhr) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java:597]: Please remove `BaseIcebergConnectorSmokeTest#getTableComment`

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java [hunk 7]
```diff
@@ -189,6 +189,7 @@ protected QueryRunner createQueryRunner()
         return queryRunner;
     }
 
+    @SuppressWarnings("DuplicateBranchesInSwitch")
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java:192]: This change is unrelated to "Support for comments on views for Delta Lake". I would recommend separating a commit.

**Comment by ebyhr** [comment]
@cla-bot check

**Comment by ebyhr** [comment]
/test-with-secrets sha=3559e1e2a8df3d04f6d3abe55204e71826338715

**Comment by ebyhr** [comment]
Could you confirm CI failure? 
```
Error:  io.trino.plugin.deltalake.TestDeltaLakeConnectorTest.testCommentView  Time elapsed: 0.783 s  <<< FAILURE!
java.lang.AssertionError: Expected query to fail: COMMENT ON VIEW test_comment_view_s914dfc9cq IS 'new comment' [QueryId: 20230427_101823_00791_ndcm9]
```

**Comment by ebyhr** [comment]
`TestDeltaLakeConnectorTest.testCommentViewColumn` is still failing. Please ping me once you fixed the CI failure.

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
/test-with-secrets sha=769a9273860a1e2802efa7af00d1bf72bee763d2
